### PR TITLE
Pin .NET MAUI Version to v6.0.312

### LINKED ISSUE #424: [CI/CD] Pin .NET MAUI Version
# [CI/CD] Pin .NET MAUI Version

In the CI pipeline, let's see if it's possible to pin the version of .NET MAUI to today's GA version, [.NET MAUI v6.0.302](https://www.nuget.org/packages/Microsoft.Maui.Dependencies/6.0.312).

Similar to how we keep the [Xamarin.CommunityToolkit NuGet Package](https://www.nuget.org/packages/Xamarin.CommunityToolkit) pinned to a lower version of Xamarin.Forms than the current stable version of Xamarin.Forms, we should pin CommunityToolkit.Maui's .NET MAUI Dependency to .NET MAUI v6.0.302 for backwards compatibility which enables our library to work for as many users as possible.

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
### Description of Change ###

 

 ### Linked Issues ###
 

 - Fixes #424 

 ### PR Checklist ###
 
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pull/XYZ 

 ### Additional information ###

### CODE REVIEW

File: samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj
Comment (brminnick) [inline samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj:4]: What do you think about __not__ adding this to the Sample App?

I'd have to test this theory, but I  think it would allow us to ensure that the Sample App continues to work with newer versions of .NET MAUI.

I.e. If I have .NET MAUI v6.0.400 installed on my machine, it would build + run the sample app using v6.0.400, right?

Maybe we could also add this scenario to the CI workflow to ensure that newer versions of .NET MAUI successfully compile and are still compatible with `CommunityToolkit.Maui`?

**Review by brminnick — COMMENTED** [review]
Thanks Vlad!! All of this looks great 💯 

I just had a thought about leveraging the Sample App to ensure that `CommunityToolkit.Maui` remains compatible with newer versions of .NET MAUI.

File: samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj
Comment (VladislavAntonyuk) [reply samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj:4]: Well, this is used in case you have multiple workloads installed on your device. 
On the one hand we can remove it at all because we set specific MAUI version on ci. 
On the other hand if we use different versions for development and testing we may receive unexpected behavior.

Comment (brminnick) [reply samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj:4]: Yup! My thought is that by removing it we can test newer versions of MAUI with the Sample App and then we'll know if the .NET MAUI Community Toolkit is working with the newest version of .NET MAUI instead of relying on a user to open an Issue.

Comment (VladislavAntonyuk) [reply samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj]: I've just updated MAUI to 6.0.400 and was not able to build the project. Seems like this tag pins .NET MAUI version and looks for exactly this version installed.
But because we pin it on yaml, we should not break development on a local devices. If build fails, we'll need to upgrade .NET MAUI

**Review by brminnick — APPROVED** [review]
Hey Vlad! 

I created two new jobs in the CI Pipeline, `build_windows_sample` + `build_macos_sample`, that install the latest .NET 6 SDK and the latest .NET MAUI Workload and then builds the sample app.

This accomplishes two things:
- Ensures the Sample App works on the latest version of .NET MAUI
- Shortens our build pipeline times
  - Instead of building the Sample App serially in the `build_windows_library` step, it is now built in parallel 

This'll help give us a heads-up if the Toolkit breaks on an upcoming .NET MAUI update.

I'll defer to you to merge this PR as long as you're cool with keeping these updates to the build pipeline!

**Pull request merged**
